conf/modules.config

```
@@ -1,32 +1,350 @@
     Available keys to override module options:
-        args            = Additional arguments appended to command in module.
-        args2           = Second set of arguments appended to command in module (multi-tool modules).
-        args3           = Third set of arguments appended to command in module (multi-tool modules).
-        publish_dir     = Directory to publish results.
-        publish_by_meta = Groovy list of keys available in meta map to append as directories to "publish_dir" path
-                            If publish_by_meta = true                 - Value of ${meta['id']} is appended as a directory to "publish_dir" path
-                            If publish_by_meta = ['id', 'custompath'] - If "id" is in meta map and "custompath" isn't then "${meta['id']}/custompath/"
-                                                                        is appended as a directory to "publish_dir" path
-                            If publish_by_meta = false / null         - No directories are appended to "publish_dir" path
-        publish_files   = Groovy map where key = "file_ext" and value = "directory" to publish results for that file extension
-                            The value of "directory" is appended to the standard "publish_dir" path as defined above.
-                            If publish_files = null (unspecified)     - All files are published.
-                            If publish_files = false                  - No files are published.
-        suffix          = File name suffix for output files.
+        ext.args   = Additional arguments appended to command in module.
+        ext.args2  = Second set of arguments appended to command in module (multi-tool modules).
+        ext.args3  = Third set of arguments appended to command in module (multi-tool modules).
+        ext.prefix = File name prefix for output files.
 ----------------------------------------------------------------------------------------
 */
+process {
+    publishDir = [
+        path: { "${params.outdir}/${task.process.tokenize(':')[-1].tokenize('_')[0].toLowerCase()}" },
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> filename.equals('versions.yml') ? null : filename },
+        enabled: false
+    ]
 
-params {
-    modules {
-        'fastqc' {
-            args = "--quiet"
-        }
-        'multiqc' {
-            args = ""
-        }
+    withName: SAMPLESHEET_CHECK {
+        publishDir = [
+            path: { "${params.outdir}/pipeline_info" },
+            mode: params.publish_dir_mode,
+            saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
+        ]
+    }
+
+    withName: FASTQC {
+        ext.args = '--quiet'
+    }
+
+    withName: TRIMGALORE {
+        ext.args = [
+            '--fastqc',
+            params.trim_nextseq > 0 ? "--nextseq ${params.trim_nextseq}" : ''
+        ].join(' ').trim()
```

```suggestion
        ext.args = { [
            '--fastqc',
            params.trim_nextseq > 0 ? "--nextseq ${params.trim_nextseq}" : ''
        ].join(' ').trim() }
```
It's a good idea generally to code all `ext.*` assignments as closures so that if users pass parameters using `-c` instead of command-line or `-params-file`, the config resolves correctly.